src/shared/sleep-config.c

```diff
@@ -127,6 +137,50 @@ int parse_sleep_config(SleepConfig **ret_sleep_config) {
         return 0;
 }
 
+static int acquire_bus(sd_bus **bus) {
+        int r;
+
+        assert(bus);
+
+        if (*bus)
+                return 0;
+
+        r = bus_connect_transport(arg_transport, arg_host, false, bus);
+        if (r < 0)
+                return bus_log_connect_error(r, arg_transport);
+
+        (void) sd_bus_set_allow_interactive_authorization(*bus, arg_ask_password);
+
+        return 0;
+}
+
+/* Freeze or thaw user.slice */
+int freeze_thaw_user_slice(const char **method) {
+        _cleanup_(bus_wait_for_units_freep) BusWaitForUnits *w = NULL;
```

Comment:
Return immediately if method is null

Reply:
probably adding assert would be safe then ?

Reply:
returning instead allows to set the cleanup var to null to skip it

Reply:
true